### PR TITLE
Add building data for NISP

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR adds Nextpnr Interchange Site Preprocessor (NISP) utility (https://github.com/antmicro/nextpnr-fpga-interchange-site-preprocessor) as a submodule and uses it to generate supplementary data to FPGA interchange device resources.

Ultimately the data will be included in BBA and BIN files for nextpnr but for now it is stored in separate JSON files.

### CODE REVIEW

**Comment by mkurc-ant** [comment]
@kboronski-ant Please take a look

File: .github/workflows/upload.yml [hunk 1]
```diff
@@ -47,13 +49,18 @@ jobs:
...
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
     - name: Save timestamp
       id: timestamp
       run: |
         echo TIMESTAMP="$(date +'%Y%m%d-%H%M%S')" >> $GITHUB_ENV
 
     - name: Generate interchange data
       run: |
+        source "$HOME/.cargo/env"
```
[the remark this answers is not in the document]
Comment (mkurc-ant) [reply .github/workflows/upload.yml]: Fixed.

File: Makefile [hunk 2]
```diff
@@ -26,11 +27,23 @@ update:
...
+	@$(IN_CONDA_ENV) cd ${NISP_PATH} && (FPGA_INTERCHANGE_SCHEMA_DIR=third_party/fpga-interchange-schema cargo build --release)
+	@ln -s `realpath ${NISP_PATH}/target/release/nisp` ${NISP_PATH}/nisp
+
+.PHONY: nisp
+nisp: ${NISP_PATH}/nisp
+
+.PHONY: clean-nisp
+clean-nisp:
+	@rm -rf ${NISP_PATH}/target
+	@unlink ${NISP_PATH}/nisp || true
```
[the remark this answers is not in the document]
Comment (mkurc-ant) [reply Makefile]: I didn't know about such syntax but now I do :) I corrected the Makefile.

**Comment by rw1nkler** [comment]
LGTM, merge if green

**Pull request merged**